addons/binding/org.openhab.binding.bosesoundtouch10/src/main/java/org/openhab/binding/bosesoundtouch10/handler/BoseSoundTouch10Handler.java

```diff
@@ -0,0 +1,378 @@
+                    response = wsHelper.get("/volume");
+                    if (response.equals("error")) {
+                        break;
+                    }
+                    String volume = xmlHelper.parsePath("/volume/actualvolume", response);
+                    if (!volume.isEmpty()) {
+                        updateState(BoseSoundTouch10BindingConstants.CHANNEL_VOLUME, new PercentType(volume));
+                    }
+                }
+                if (command instanceof PercentType) {
+                    PercentType num = (PercentType) command;
+                    response = wsHelper.setVolume(num);
+                }
+                break;
+            case BoseSoundTouch10BindingConstants.CHANNEL_BASS:
+                if (command instanceof RefreshType) {
+                    response = getBassAndUpdateUI();
+                    if (response.equals("error")) {
+                        break;
+                    }
+                }
+                if (command instanceof DecimalType) {
+                    DecimalType num = (DecimalType) command;
+                    response = wsHelper.setBass(num);
+                }
+                break;
+            case BoseSoundTouch10BindingConstants.CHANNEL_CONTROL:
+                if (command instanceof PlayPauseType) {
+                    PlayPauseType playPause = (PlayPauseType) command;
+                    switch (playPause.toString()) {
+                        case "PLAY":
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PLAY);
+                            break;
+                        case "PAUSE":
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PAUSE);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            case BoseSoundTouch10BindingConstants.CHANNEL_SOURCE:
+                if (command instanceof RefreshType) {
+                    response = wsHelper.get("/now_playing");
+                    if (response.equals("error")) {
+                        break;
+                    }
+                    String source = xmlHelper.parseSource("/nowPlaying", response, "source");
+                    if (source.equals(Sources.STANDBY.toString())) {
+                        updateState(BoseSoundTouch10BindingConstants.CHANNEL_POWER, OnOffType.OFF);
+                        updateState(BoseSoundTouch10BindingConstants.CHANNEL_NOW_PLAYING, new StringType(""));
+                    } else {
+                        String sourceName = xmlHelper.parsePath("/nowPlaying/ContentItem/itemName", response);
+                        response = wsHelper.get("/presets");
+                        if (response.equals("error")) {
+                            break;
+                        }
+                        switch (source) {
+                            case "BLUETOOTH":
```

Comment:
Please put constants somewhere for those fixed Strings. "BLUETOOTH", "AUX", "INTERNET_RADIO" etc.